sormas-ui/src/main/java/de/symeda/sormas/ui/events/EventController.java

```diff
@@ -207,6 +255,37 @@ public void onCommit() {
 		return editView;
 	}
 
+	public CommitDiscardWrapperComponent<EventDataForm> getEventCreateComponent(ContactDto contact) {
+
+		EventDataForm eventCreateForm = new EventDataForm(true, false);
+		eventCreateForm.setValue(createNewEvent(contact.getDisease()));
+		eventCreateForm.getField(EventDto.DISEASE).setReadOnly(true);
+
+		final CommitDiscardWrapperComponent<EventDataForm> editView = new CommitDiscardWrapperComponent<>(
+			eventCreateForm,
+			UserProvider.getCurrent().hasUserRight(UserRight.EVENT_CREATE),
+			eventCreateForm.getFieldGroup());
+
+		editView.addCommitListener(() -> {
+			if (!eventCreateForm.getFieldGroup().isModified()) {
+				EventDto dto = eventCreateForm.getValue();
+				FacadeProvider.getEventFacade().saveEvent(dto);
+				Notification.show(I18nProperties.getString(Strings.messageEventCreated), Type.WARNING_MESSAGE);
+
+				if (contact.getCaze() != null) {
+					EventReferenceDto createdEvent = new EventReferenceDto(dto.getUuid());
+
+					createEventParticipantWithContact(createdEvent, contact);
```

Comment:
What's the reason for only doing this when the contact has a source case? What about contacts without a source case?

Reply:
Thanks for pointing that out ! That's on me, it's an artifact of when I first c/c the equivalent method for Cases and quickly changed thinks to get a compiling one for Contacts. I forgot to adapt this part. Fixed it !

It's there because it's Cases' method that's used (with null as case param) even for the base event creation (hence the https://github.com/hzi-braunschweig/SORMAS-Project/pull/3371#discussion_r519666907). But it feels a bit cranky. I think I should refactor this and add a proper create method for it if you don't mind it.